### PR TITLE
feat(ImageBlock): allow attachments when block is empty

### PR DESCRIPTION
**Pull request by fulopdaniel**
https://www.figma.com/file/pYt8ECuB6Sc63cVtpKiDgf/Blocks---Updated?type=design&node-id=485-128059&mode=design&t=Ax3uS4Y06OpRKh6h-0

https://www.loom.com/share/911b7fb9330b4b9a8ac98dd2aae8a03f

CU-86947yy57

https://app.clickup.com/t/86947yy57

## Changes

1. Attachments moved to parent component so overflow:hidden is not applied to that.
2. Added a 44px min height to the image component if there are attachments.
3. Hide captions if there is none in view mode.

### CODE REVIEW

File: packages/image-block/src/components/ImageCaption.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {
 } from '@frontify/guideline-blocks-settings';
 import { getCaptionPlugins, titlePlugins } from './helpers';
 import { AppBridgeBlock } from '@frontify/app-bridge';
-import { CaptionPosition, type Settings } from '../types';
+import { CaptionPosition, Ratio, type Settings, textRatioValues } from '../types';
```
Comment (ragi96) [inline packages/image-block/src/components/ImageCaption.tsx:12]:
```suggestion
import { CaptionPosition, type Ratio, type Settings, textRatioValues } from '../types';
```

**Review by ragi96 — COMMENTED** [review]
Also add tests at least for these cases:

1. Attachment set - no asset -> Attachment Shown
2. Attachment set - no asset - download active -> Attachment Shown without Download Button

**Comment by fulopdaniel** [comment]
> Screenshot 2024-04-05 at 13 49 01
> 
> Displays the download button without an asset

Nice catch, fixed, thanks! Added a testcase for it. (Merged the two test case suggestions you made since they are essentially the same: hide download, show attachments in view mode.)

File: packages/image-block/src/components/ImageCaption.tsx [hunk 2]
```diff
@@ -30,36 +31,47 @@ export const ImageCaption = ({
     description,
     name,
     positioning,
+    ratio,
     setBlockSettings,
 }: ImageCaptionProps) => {
-    return (
+    const hasTitle = hasRichTextValue(name);
+    const hasDescription = hasRichTextValue(description);
+    return isEditing || hasTitle || hasDescription ? (
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/image-block/src/components/ImageCaption.tsx]: Sure, adjusted

File: packages/image-block/src/components/ImageCaption.tsx (hunk 1 above)
Comment (fulopdaniel) [reply packages/image-block/src/components/ImageCaption.tsx]: fixed

**Review by ragi96 — APPROVED** [review]
LGTM

**Pull request merged**